api/events/schema.py

```diff
@@ -69,6 +70,31 @@ def mutate(self, info, **kwargs):
         return CancelEvent(event=event)
 
 
+class EndEvent(graphene.Mutation):
+    """
+    Mutation to end an event
+    Returns event payload on ending the event
+    """
+    class Arguments:
+        calendar_id = graphene.String(required=True)
+        event_id = graphene.String(required=True)
+        start_time = graphene.String(required=True)
+        end_time = graphene.String(required=True)
+        meeting_end_time = graphene.String(required=True)
+    event = graphene.Field(Events)
+
+    def mutate(self, info, **kwargs):
+        room_id, event = check_event_in_db(self, info, "ended", **kwargs)
+        if not event:
+            event = EventsModel(
+                event_id=kwargs['event_id'],
+                meeting_end_time=kwargs['meeting_end_time']
+                )
+            event.save()
+
```

Comment:
@ThaDeveloper 
Why you are creating the event if it is not in the db? 
since what is required is to end the event.

Reply:
@rajeman What happening here is in case the `check_event_in_db` doesn't return an event we will try to save the event and since the id doesn't exist it will throw an error.

Reply:
@ThaDeveloper  I need more clarification :
 if a user tries to end an event and the event is not in the db, what is the update for?

Reply:
@rajeman It's simply for validation, means they tried to end an event that doesn't exist and calling save on the event model does that. Also, refer to `CancelEvent` and `Checkin`

Reply:
@ThaDeveloper 
I get you now. The confusion is from the name of that method  `check_event_in_db`. The name doesn't indicate it also tries to end an event